kubernetes_state/check.py

```diff
@@ -217,17 +223,37 @@ def kube_job_complete(self, message, **kwargs):
         for metric in message.metric:
             tags = []
             for label in metric.label:
-                tags.append(self._format_tag(label.name, label.value))
+                trimmed_job = self._trim_job_tag(label.value)
+                tags.append(self._format_tag(label.name, trimmed_job))
```

Comment:
Last thing before merge: the good tag logic was the one you deleted: you should:

- iterate on labels
- if label == `job`, trim, format and append
- else format and append

If your version of the code, if a `namespace` value matches the pattern, it will be incorrectly trimmed.